core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -909,32 +922,49 @@ public boolean load() {
         }
     }
 
-    private boolean loadBaseGraphAndProperties() {
-        if (properties.loadExisting()) {
-            if (properties.containsVersion())
-                throw new IllegalStateException("The GraphHopper file format is not compatible with the data you are " +
-                        "trying to load. You either need to use an older version of GraphHopper or run a new import");
-            // check encoding for compatibility
-            String flagEncodersStr = properties.get("graph.flag_encoders");
-
-            if (!encodingManager.toFlagEncodersAsString().equalsIgnoreCase(flagEncodersStr)) {
-                throw new IllegalStateException("Flag encoders do not match:"
-                        + "\nGraphhopper config: " + encodingManager.toFlagEncodersAsString()
-                        + "\nLoaded graph:       " + flagEncodersStr
-                        + "\nChange configuration to match the graph or delete " + baseGraph.getDirectory().getLocation());
-            }
+    private void loadEncodingManagerFromProperties(StorableProperties properties) {
+        if (properties.containsVersion())
+            throw new IllegalStateException("The GraphHopper file format is not compatible with the data you are " +
+                    "trying to load. You either need to use an older version of GraphHopper or run a new import");
+
+        String versionStr = properties.get("graph.em.version");
+        if (versionStr.isEmpty() || !String.valueOf(Constants.VERSION_EM).equals(versionStr))
+            throw new IllegalStateException("Incompatible encoding version. You need to use the same GraphHopper version you used to import the graph, or run a new import. "
+                    + " Stored encoding version: " + (versionStr.isEmpty() ? "missing" : versionStr) + ", used encoding version: " + Constants.VERSION_EM);
```

Comment:
We have this new `Constants.VERSION_EM` now that defines the version of the stored encoding manager. We do not need a 'version' for each encoded value separately I think, because we store all the different encoded values with all their properties.